tests/unit/models/test_project.py

```diff
@@ -181,7 +156,6 @@ def load_project_yaml(yaml_loaded_data) -> Project:
         IMAGECRAFT_YAML_GENERIC,
         IMAGECRAFT_YAML_SIMPLE_PLATFORM,
         IMAGECRAFT_YAML_MINIMAL_PLATFORM,
-        IMAGECRAFT_YAML_NO_BUILD_FOR_PLATFORM,
```

Comment:
This should still work.  At least it still works for snapcraft.  

Or is this an error that only occurs in the unit tests but still works in when using imagecraft itself?

Reply:
Yeah, you're right - I think https://github.com/canonical/craft-application/pull/735 fixes this.

Reply:
I understand that now `build-for` cannot be missing if `build-on` is specified. I also saw tests for that in craft-application. Do you confirm? Am I missing something?